react-graph/src/Line.tsx

```diff
@@ -42,119 +43,113 @@ export interface IProps {
             } catch {
-            setHighlight([NaN, NaN]);
+                setHighlight([NaN, NaN]);
             }
         }
-   }, [data, context.XHover])
+    }, [data, context.XHover])
 
     React.useEffect(() => {
-        if (props.data == null || props.data.length === 0) setData(null);
+        if (props.data == null) setData(null);
         else setData(new PointNode(props.data));
-    },[props.data]);
-
-   React.useEffect(() => {
-       if (guid === "")
-           return;
-       context.SetLegend(guid, createLegend());
-
-   }, [highlight, enabled]);
+    }, [props.data]);
 
-   React.useEffect(() => {
-        if (data == null) {
-            setVisibleData([]);
+    React.useEffect(() => {
+        if (guid === "")
             return;
-        }
-        setVisibleData(data.GetData(context.XDomain[0],context.XDomain[1],true));
-    },[data, context.XDomain[0], context.XDomain[1]])
+        context.SetLegend(guid, createLegend());
+
+    }, [highlight, enabled]);
 
     React.useEffect(() => {
         const id = context.AddData(createContextData());
         setGuid(id);
         return () => { context.RemoveData(id) }
     }, []);
 
-   function generateData() {
-       let result = "M ";
-       if (data == null)
-        return ""
-     result = result + visibleData.map((pt, _) => {
-           const x = context.XTransformation(pt[0]);
-           const y = context.YTransformation(pt[1], AxisMap.get(props.axis));
-           return `${x},${y}`
-       }).join(" L ")
-
-       return result
-   }
-
-
-   return (
-       enabled?
-       <g>
-           <path d={generateData()} style={{ fill: 'none', strokeWidth: props.width === undefined ? 3 : props.width, stroke: props.color }} strokeDasharray={LineMap.get(props.lineStyle)} />
-           {showPoints && data != null? visibleData.map((pt, i) => <circle key={i} r={3} cx={context.XTransformation(pt[0])} cy={context.YTransformation(pt[1], AxisMap.get(props.axis))} fill={props.color} stroke={'currentColor'} style={{ opacity: 0.8/*, transition: 'cx 0.5s,cy 0.5s'*/ }} />) : null}
-           {(props.highlightHover ?? false) && !isNaN(highlight[0]) && !isNaN(highlight[1])? 
-            <circle r={5} cx={context.XTransformation(highlight[0])} cy={context.YTransformation(highlight[1], AxisMap.get(props.axis))} fill={props.color} stroke={'currentColor'} style={{ opacity: 0.8/*, transition: 'cx 0.5s,cy 0.5s'*/ }} /> : null}
-       </g > : null
-   );
-}
-
-export default Line;
+    const generateData = React.useCallback((pointsToDraw: number[][]) => {
+        let result = "M ";
+        if (data == null)
+            return ""
+
+        result = result + pointsToDraw.map((pt, _) => {
+            const x = context.XTransformation(pt[0]);
+            const y = context.YTransformation(pt[1], AxisMap.get(props.axis));
+            return `${x},${y}`
+        }).join(" L ")
+
+        return result
+    }, [context.XTransformation, context.YTransformation])
```

Comment:
```suggestion
    }, [context.XTransformation, context.YTransformation, propa.axis])
```

otherwhise this may cause issues when the axis is changed (e.g. in PQBrowser Trend)